pkg/handler/handler.go

```
@@ -83,6 +83,19 @@ func (h *handler) queryMetricData(ctx context.Context, req *backend.QueryDataReq
 	return &backend.QueryDataResponse{Responses: responses}, nil
 }
 
+// waitForQueries blocks until every query has reported back, or until the
+// request context is cancelled because the user navigated away.
+func waitForQueries(ctx context.Context, done <-chan struct{}, n int) {
+	for i := 0; i < n; i++ {
+		select {
+		case <-done:
+		case <-ctx.Done():
+			backend.Logger.Debug("Request cancelled, stopping wait")
+			break
```

golangci-lint (2.12.2)

[error] SA4011: ineffective break statement. Did you mean to break out of the outer loop?

(staticcheck)